### PR TITLE
objectives and design choices added

### PR DESCRIPTION
**Pull request by gertingold**
This PR proposes an implementation of the ideas of @pdebuyl in #434.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -3,11 +3,30 @@ Contributing
...
+  Currently, pandas, statmodels, seaborn, scikit-image, scikit-learn,
+  sympy, and mayavi are discussed.
+* Wherever possible, automated testing is applied to the code examples.
+
+Design choices:
+
+* Each chapter should provide a useful basis for a 1‒2 h tutorial.
+* The code should be readable.
+* An idomatic style should be followed, e.g. ``import numpy as np``,
+  preference for array operations.
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
  preference for array operations, PEP8 coding conventions.
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -3,11 +3,30 @@ Contributing
...
+Contributors should keep the following objectives and design choices of
+the SciPy lecture notes in mind.
+
+Objectives:
+
+* The SciPy lecture notes provide a self-contained introduction to Python
+  and its primary computational packages, the ”SciPy stack“.
+* They provide an introduction to a selection of computational libraries.
+  Currently, pandas, statmodels, seaborn, scikit-image, scikit-learn,
+  sympy, and mayavi are discussed.
```
[the remark this answers is not in the document]
Comment (gertingold) [reply CONTRIBUTING.rst]: Do you have specific points which you would like to see here?

**Comment by GaelVaroquaux** [comment]
> Do you have specific points which you would like to see here?

I would rather hear how others (you?) think about this tradeoff. I do not
want to start by imposing my views.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -3,11 +3,30 @@ Contributing
...
+* Wherever possible, automated testing is applied to the code examples.
```
[the remark this answers is not in the document]
Comment (gertingold) [reply CONTRIBUTING.rst]: fixed

File: CONTRIBUTING.rst (hunk 1 above)
Comment (gertingold) [reply CONTRIBUTING.rst]: fixed

**Comment by gertingold** [comment]
Ok, I'll have to think a bit about it, and may be there will be input from others. @pdebuyl, @vahtras, ...?

**Comment by vahtras** [comment]
Not sure what you mean by trade-off here.

**Comment by pdebuyl** [comment]
The tradeoff means, at least that is what I understand, that we should refrain from including too many packages (there is a maintaining cost to every chapter here). On the other hand, the breadth of the lecture notes makes them more appealing and useful.

**Comment by gertingold** [comment]
Criteria could be for example:
* maturity of a package (which would contribute to keeping maintenance cost low)
* widespread use in the scientific Python community

**Pull request merged**